writer.go

```diff
@@ -364,7 +384,12 @@ func diffp(new []int, old []int) (diff []int) {
 	return
 }
 
-type writer struct {
+type PartitionWriter interface {
```

Comment:
Same comment as for NewPartitionWriter in the configuration object, if all we're solving with this abstraction is to mock tests let's not make it part of the public API.